sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -124,7 +124,15 @@ case class SortArray(base: Expression, ascendingOrder: Expression)
 
   override def checkInputDataTypes(): TypeCheckResult = base.dataType match {
     case ArrayType(dt, _) if RowOrdering.isOrderable(dt) =>
-      TypeCheckResult.TypeCheckSuccess
+      ascendingOrder match {
+        case Literal(_: Boolean, BooleanType) =>
+          TypeCheckResult.TypeCheckSuccess
+        case _ =>
+          TypeCheckResult.TypeCheckFailure(
+            s"Sort order in second argument requires ${BooleanType.simpleString} type as " +
```

Comment:
Actually, I wrote this as I was worried of the case `cast(NULL as boolean)`.

> Sort order in second argument requires boolean type as non-null, however, it is 'CAST(NULL AS BOOLEAN)' as boolean type.

I will just make it short. Actually, that sentence was copied from the default one.